addons/mrp/models/mrp_production.py

```diff
@@ -946,22 +950,21 @@ def _render_note_exception_quantity_mo(rendering_context):
                 'production_order': self,
                 'order_exceptions': dict((key, d[key]) for d in rendering_context for key in d),
                 'impacted_pickings': False,
-                'cancel': False
+                'cancel': cancel
             }
             return self.env.ref('mrp.exception_on_mo').render(values=values)
-
         documents = {}
-        for move, (old_qty, new_qty) in moves_modification.items():
-            document = self.env['stock.picking']._log_activity_get_documents(
-                {move: (old_qty, new_qty)}, 'move_dest_ids', 'DOWN', _keys_in_sorted, _keys_in_groupby)
-            for key, value in document.items():
-                if documents.get(key):
-                    documents[key] += [value]
-                else:
-                    documents[key] = [value]
+        document = self.env['stock.picking']._log_activity_get_documents(moves_modification, 'move_dest_ids', 'DOWN', _keys_in_sorted, _keys_in_groupby)
+        if not document:
+            document = self.env['stock.picking']._less_quantities_than_expected_add_documents(moves_modification, {})
```

Comment:
@amoyaux : All changes are done. Except this. As it is generating the python error for dict which is passing to ['stock.picking']._log_activity() as - 
{(mrp.production(60,), res.users(1,)): {stock.move(342,): (stock.move(345,), (2.0, 0.0))}}
instead of 
defaultdict(<class 'list'>, {(mrp.production(64,), res.users(1,)): [{stock.move(354,): (stock.move(357,), (2.0, 0.0))}]})

Please let me know your concern.